components/events/src/event_bus.cpp

```diff
@@ -48,8 +45,7 @@ void EventBus::process()
 	{
 		if ((*it).expired())
 		{
-			m_observers.erase(it);
-			it++;
+			it = m_observers.erase(it);
```

Comment:
Here, `it` is already advanced to the next observer.
That is, you should move the `it++` from the loop to the bottom of its body:
```
for ( auto it = m_observers.begin(); it != m_observers.end();)
{
    ....
    ++it;
}
```

Reply:
Ill need to add a test for this

Reply:
Yep, this was the correct fix and the test confirms this! see `expire an observer before process` in `events_bus.test.cpp`